workflows/prognostic_c48_run/runtime/overrider.py

```diff
@@ -0,0 +1,130 @@
+import dataclasses
+import functools
+import logging
+from typing import Hashable, Mapping, MutableMapping, Set
+
+import cftime
+import fsspec
+import xarray as xr
+
+import fv3gfs.util
+from runtime.monitor import Monitor
+from runtime.types import Diagnostics, Step
+from runtime.derived_state import DerivedFV3State
+from runtime.utils import quantity_state_to_ds, ds_to_quantity_state
+
+QuantityState = MutableMapping[Hashable, fv3gfs.util.Quantity]
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class OverriderConfig:
+    """Configuration for overriding tendencies from a step.
+    
+    Attributes:
+        url: path to dataset containing tendencies.
+        variables: mapping from state name to name of corresponding tendency in
+            provided dataset. For example: {"air_temperature": "fine_res_Q1"}.
+    """
+
+    url: str
+    variables: Mapping[str, str]
+
+
+@dataclasses.dataclass
+class OverriderAdapter:
+    """Wrap a Step function to override certain tendencies."""
+
+    config: OverriderConfig
+    state: DerivedFV3State
+    communicator: fv3gfs.util.CubedSphereCommunicator
+    timestep: float
+    diagnostic_variables: Set[str] = dataclasses.field(default_factory=set)
+
+    def __post_init__(self: "OverriderAdapter"):
+        if self.communicator.rank == 0:
+            logger.debug(f"Opening tendency overriding dataset from: {self.config.url}")
+        tile = self.communicator.partitioner.tile_index(self.communicator.rank)
+        mapper = fsspec.get_mapper(self.config.url)
+        ds = xr.open_zarr(mapper, consolidated=True)
+        ds = ds[list(self.config.variables.values())].isel(tile=tile)
+        self._tendency_ds = DatasetCachedByChunk(ds, "time")
```

Comment:
Did you try using [LRUStoreCache](https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.LRUStoreCache)? Seems like it could be one-liner, and if it achieves similar performance I would opt for that over rolling our own custom caching layer.
```suggestion
        mb = 2**24
        mapper = zarr.storage.LRUStoreCache(fsspec.get_mapper(self.config.url), max_size=(128 * mb)
        ds = xr.open_zarr(mapper, consolidated=True)
        ds = ds[list(self.config.variables.values())].isel(tile=tile)
        self._tendency_ds = ds
```

Reply:
good call, i tried it previously but for some reason couldn't get it working. seems to do the trick nicely now.